decoder/src/frame.rs

```diff
@@ -9,6 +9,35 @@ use chrono::TimeZone;
 use colored::Colorize;
 use defmt_parser::{DisplayHint, Fragment, Level, ParserMode, TimePrecision, Type};
 
+/// Used to convert a `i128` value into right target type in hex
+struct Hex128(i128, Type);
+
+impl std::fmt::LowerHex for Hex128 {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.1 {
+            Type::I8 => fmt::LowerHex::fmt(&(self.0 as i8), f),
```

Comment:
What about unsigned types? If not required, we should add a comment to explain why they are missing.

Reply:
The struct `Hex128` is renamed to `I128Hex` to indicate this applies only to signed integers. Unsigned integers are formatted correctly as hex numbers. The snapshot test `log` also outputs a line with unsigned integers.